app/models/miq_server/log_management.rb

```diff
@@ -147,55 +118,56 @@ def pg_log_patterns
+  def log_start_and_end_for_pattern(pattern)
+    evm = VMDB::Util.get_evm_log_for_date(pattern)
+    return if evm.nil?
 
-      log_prefix = "Task: [#{task.id}]"
-      msg = "Posting logs for: #{resource}"
-      _log.info("#{log_prefix} #{msg}")
-      task.update_status("Active", "Ok", msg)
+    VMDB::Util.get_log_start_end_times(evm)
+  end
 
-      msg = "Zipping and posting current logs and configs on #{resource}"
-      _log.info("#{log_prefix} #{msg}")
-      task.update_status("Active", "Ok", msg)
+  def post_one_log_pattern(pattern, logfile, log_type)
+    task = logfile.miq_task
+    log_prefix = "Task: [#{task.id}]"
 
-      local_file = VMDB::Util.zip_logs("evm.zip", current_log_patterns, "system")
+    log_start, log_end = log_start_and_end_for_pattern(pattern)
+    date_string = "#{format_log_time(log_start)} #{format_log_time(log_end)}" unless log_start.nil? && log_end.nil?
 
-      evm = VMDB::Util.get_evm_log_for_date("log/*.log")
-      log_start, log_end = VMDB::Util.get_log_start_end_times(evm)
+    msg = "Zipping and posting #{log_type.downcase} logs for [#{who_am_i}] from: [#{log_start}] to [#{log_end}]"
+    _log.info("#{log_prefix} #{msg}")
+    task.update_status("Active", "Ok", msg)
 
-      date_string = "#{format_log_time(log_start)} #{format_log_time(log_end)}" unless log_start.nil? && log_end.nil?
-      name = "Requested #{self.name} logs #{date_string} "
-      desc = "Logs for Zone #{zone.name rescue nil} Server #{self.name} #{date_string}"
+    begin
+      local_file = VMDB::Util.zip_logs("evm.zip", log_patterns(log_type, pattern), "system")
+      self.log_files << logfile
 
       logfile.update_attributes(
-        :file_depot         => log_depot,
         :local_file         => local_file,
         :logging_started_on => log_start,
         :logging_ended_on   => log_end,
-        :name               => name,
-        :description        => desc,
+        :name               => logfile_name(log_type, date_string),
+        :description        => "Logs for Zone #{zone.name rescue nil} Server #{self.name} #{date_string}",
       )
 
       logfile.upload
+
     rescue StandardError, Timeout::Error => err
-      _log.error("#{log_prefix} Posting of current logs failed for #{resource} due to error: [#{err.class.name}] [#{err}]")
+      _log.error("#{log_prefix} Posting of #{log_type.downcase} logs failed for #{who_am_i} due to error: [#{err.class.name}] [#{err}]")
       logfile.update_attributes(:state => "error")
       raise
     end
-    msg = "Current log files from #{resource} are posted"
+    msg = "#{log_type} log files from #{who_am_i} are posted"
     _log.info("#{log_prefix} #{msg}")
     task.update_status("Active", "Ok", msg)
   end
 
+  def post_current_logs(taskid, log_depot)
+    delete_old_requested_logs
+
+    logfile = LogFile.current_logfile
+    logfile.update(:file_depot => log_depot, :miq_task   => MiqTask.find(taskid))
+    post_one_log_pattern("log/*.log", logfile, "Current")
```

Comment:
^ this was the goal.  It's hard to see in post_historical_logs but there's some "historical log specific" setup and then the call to `post_one_log_pattern`.

For an idea of what the automate stuff will look like due to this refactoring:

```ruby
  def post_automate_models(taskid, log_depot)
    domain_zip = Rails.root.join("log/domain.zip")
    backup_automate_models(domain_zip)

    logfile = LogFile.historical_logfile
    logfile.update(:file_depot => log_depot, :miq_task => MiqTask.find(taskid))
    post_one_log_pattern(domain_zip, logfile, "Models")
  ensure
    FileUtils.rm_rf(domain_zip)
  end
```